### ISSUE TITLE
can only load extensions once

### ISSUE THREAD

**Issue by srepmub** [holoviz/panel#3547]
the following results in a warning, but did still work for me with panel 0.12. as of 0.13 I have to load the extensions in a single call to panel.extension:

```
import panel
panel.extension('plotly')
panel.extension('vtk')
```

I think this means there is now a problem when using multiple panel extensions from different providers/libraries in a single notebook, as these libraries don't know about each other..?

ideally, each library could just use statements as above, but I'm sure there's a reason that won't work? or how should I proceed here (as the author of one such library).

**Comment by philippjfr** [comment]
Adding extensions separately embeds all of bokeh.js and panel.js each time ballooning the size of the notebook. Therefore you should always load the extensions you need in a single `pn.extension` call.

**Comment by srepmub** [comment]
thanks. the thing is, I'm providing a wrapper around panel, and users don't even know it's being used. so having a way of incrementally updating the loaded extension from the wrapper library would be much nicer..

**Comment by philippjfr** [comment]
Instead of running the extension you could just import the appropriate modules from panel, e.g. `pn.extension('plotly')` really just runs `import panel.models.plotly`. If you do that you can just call `pn.extension()` once after you've run all those imports. Here's the import mapping for the different extensions:

```python
    _imports = {
        'katex': 'panel.models.katex',
        'mathjax': 'panel.models.mathjax',
        'plotly': 'panel.models.plotly',
        'deckgl': 'panel.models.deckgl',
        'vega': 'panel.models.vega',
        'vtk': 'panel.models.vtk',
        'ace': 'panel.models.ace',
        'echarts': 'panel.models.echarts',
        'ipywidgets': 'ipywidgets_bokeh.widget',
        'perspective': 'panel.models.perspective',
        'terminal': 'panel.models.terminal',
        'tabulator': 'panel.models.tabulator',
        'texteditor': 'panel.models.quill',
        'jsoneditor': 'panel.models.json_editor'
    }
```

**Comment by srepmub** [comment]
thanks. that is good to know. I guess if two hypothetical wrapper libraries both use direct imports this way, and they each (try to) call pn.extension() before the first plot, it would work without the user having to call it..

**Comment by philippjfr** [comment]
Ah, I guess that is an issue because only the first call to extension will bundle all the contents. I'll reopen to brainstorm solution.

**Comment by philippjfr** [comment]
This was resolved in Panel 1.3.x, we now allow partial loading of the extension.
